MERC-5889:  SBRV Tests for Verifier's SetConfig

dav009 opened the pull request:
[MERC-5889](https://smartcontract-it.atlassian.net/browse/MERC-5889)

### Resolves Dependencies
- https://github.com/smartcontractkit/chainlink/pull/13813

## What ?

- Added Tests for `DestinationVerifier.SetConfig`
- Added a couple extra `get` methods to verifier contract so we can easily check expected internal state

   
   
## Notes

> forge test --match-path "src/v0.8/llo-feeds/v0.4.0/test/verifier/*"

```
Suite result: ok. 12 passed; 0 failed; 0 skipped; finished in 13.61ms (19.62ms CPU time)
```   


[MERC-5889]: https://smartcontract-it.atlassian.net/browse/MERC-5889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dav009 committed:
llo-feeds: verifier SetConfig tests

dav009 committed:
llo-feeds: adding verifier contract get methods for easier testing

dav009 commented on contracts/src/v0.8/llo-feeds/v0.4.0/test/verifier/VerifierSetConfigTest.t.sol:
```diff
@@ -0,0 +1,260 @@
+        address[] memory signerAddrs = _getSignerAddresses(signers);
+        Common.AddressAndWeight[] memory weights = new Common.AddressAndWeight[](1);
+        weights[0] = Common.AddressAndWeight(signers[0].signerAddress, ONE_PERCENT * 100);
+        s_verifier.setConfig(signerAddrs, FAULT_TOLERANCE, weights);
```
todo: check side effect here

dav009 committed:
adding rewards wire up for testing

dav009 committed:
incomeplte fix for test_setConfigWithAddressesAndWeightsAreSetCorrectly

dav009 committed:
llo-feeds: adjusting setConfig tests due to changes

Fletch153 commented on contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifier.sol:
```diff
@@ -387,5 +388,13 @@ contract DestinationVerifier is IDestinationVerifier, ConfirmedOwner, TypeAndVer
         return address(s_feeManager);
     }
+
+    function getDONConfigs() public view returns (DONConfig[] memory) {
```
We don't want to expose any configuration details that could lead to parasitic usage. By exposing this, users could get the DONConfigs and verify without paying the feee

Fletch153 commented on contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifier.sol:
```diff
@@ -387,5 +388,13 @@ contract DestinationVerifier is IDestinationVerifier, ConfirmedOwner, TypeAndVer
+        return s_DONConfigs;
+    }
+
+     function getSignerConfigByAddressAndDONConfigId(bytes32 _id) public view returns (bool) {
```
Same as above, these should all be private with no getters

dav009 committed:
llo-feeds: remove interface checks for verifierProxy

dav009 committed:
llo-feeds: clean up setConfig tests from checking internal state

dav009 commented:
@Fletch153  addressed comments in this PR:
- removed getter functions
- removed tests checking internal state
 - these will be addressed via calls to verify in another PR
